### PR TITLE
#63 - Add Redis (Backend)

### LINKED ISSUE #63: Add Redis (Backend)

### PR DESCRIPTION
**Pull request by NN171**
In this task were added Redis and Flyway migration. The application, db and Redis were dockerized into docker-compose file.

### CODE REVIEW

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/ProfileService.java [hunk 1]
```diff
@@ -10,7 +10,7 @@ public interface ProfileService {
 
     void saveProfile(List<String> properties);
 
-    List<ProfileDto> getProfilesByInstituteName(String institute);
+    List<ProfileDto> getProfilesByInstituteName(String name);
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/ProfileService.java:13]: May use the parameter name `instituteName`?

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/DirectionServiceImpl.java [hunk 2]
```diff
@@ -80,7 +84,7 @@ public void saveDirection(List<Element> props) {
             String level = props.get(2).text().trim();
             String form = props.get(3).text().trim();
 
-            if (directionRepository.findByCode(code) == null && form.equals("очная") && (level.equals("бакалавриат") || level.equals("специалитет"))) {
+            if (directionRepository.findByCode(code).isEmpty() && form.equals("очная") && (level.equals("бакалавриат") || level.equals("специалитет"))) {
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/DirectionServiceImpl.java:87]: 👍

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/ProfileServiceImpl.java [hunk 3]
```diff
@@ -41,14 +42,16 @@ public ProfileServiceImpl(ProfileRepository profileRepository,
     }
 
     @Override
-    public List<ProfileDto> getProfilesByInstituteName(String institute) {
-        List<Profile> profiles = profileRepository.findByInstitute(institute)
+    @Cacheable(value = "ProfileService::getProfilesByInstitute", key = "#name")
+    public List<ProfileDto> getProfilesByInstituteName(String name) {
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/ProfileServiceImpl.java:46]: If you will decide to rename `instituteName` in the ProfileService.java, don't forget to rename this parameter

Comment (NN171) [reply backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/ProfileServiceImpl.java:46]: Ok, thanks

**Pull request merged**